### PR TITLE
Access environment variables inside templates

### LINKED ISSUE #13: Access environment variables inside templates
First of all: Thank you very much for dotdrop!

To make it really usable for me, I need access to environment variables inside the templates.

That way - instead af having `{%@@ if profile == "PROFILENAME" @@%}` all over the place - I can just say `{{@@ env['MY_VAR'] @@}}` to set host-specific properties. I have an `.env`-file in my directory, that is not tracked by git and I invoke dotdrop with the following alias:

```
alias dotdrop='eval $(cat ~/dotfiles/.env | egrep -v "^#") ~/dotfiles/dotdrop.sh'
```

This also gives me the possibility to use dotdrop for files that need plain-text credentials or other information I don't want to share with the public.

You can see how I use it in my [dotfiles](https://github.com/open-dynaMIX/dotfiles) (beware: I should spend some time to clean it up a bit :wink:).

Since it's just two lines, that needs to be changed, for the time being, I've just edited them locally. If there is an interest in this, I can make a PR, when I find the time.

### PR DESCRIPTION
**Pull request by open-dynaMIX**
Closes #13

I've allowed myself to also clean up a tiny bit.

English is not my first language, if you find any mistakes, just point them out.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -386,6 +386,40 @@ Note that dotdrop uses different delimiters than
 * comment start = `{#@@`
 * comment end = `@@#}`
 
+## Available variables
+### Profile
+
+`{{@@ profile @@}}` contains the profile being used. Below in the example you can see how this is useful.
```
Comment (deadc0de6) [inline README.md]: *contains the profile being used. Below example shows how it is used.*

File: README.md [hunk 2]
```diff
@@ -386,6 +386,40 @@ Note that dotdrop uses different delimiters than
...
+
+### Environment variables
+
+It's possible to access environment variables inside the templates. You can utilize this functionality like this:
```
Comment (deadc0de6) [inline README.md]: ~~You can utilize this functionality like this~~ -> *This feature can be used like this*

File: README.md [hunk 3]
```diff
@@ -386,6 +386,40 @@ Note that dotdrop uses different delimiters than
...
+
+```
+{{@@ env['MY_VAR'] @@}}
+```
+
+This allows for storing host-specific properties and/or secrets as environment variables.
```
Comment (deadc0de6) [inline README.md]: *... secrets in environment ...*

**Comment by deadc0de6** [comment]
Awesome !

As soon as the English fixes are done, I'll merge it.

Thanks again for your contribution!

BTW I was planning on adding a section with pointers to repositories where dotdrop is used, would you be ok if I add yours ?

**Comment by open-dynaMIX** [comment]
> Thanks again for your contribution!

Glad to be of help :smile:

> BTW I was planning on adding a section with pointers to repositories where dotdrop is used, would you be ok if I add yours ?

Absolutely!

**Comment by open-dynaMIX** [comment]
Sorry, I accidentally commited as a different author (different computer). So I had to change that.

**Pull request merged**